### PR TITLE
[MLOP-635] Rebase Incremental Job/Interval Run branch for test on selected feature sets

### PR DESCRIPTION
**Pull request by moromimay**
## Why? :open_book:
We wanted to optimize Butterfree's processing capabilities.

## What? :wrench:
This PR intends to introduce interval processing to Butterfree.

## Type of change
Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How everything was tested? :straight_ruler:
Unit and integration tests.

## Checklist
- [x] My code follows the style guidelines of this project (docstrings, type hinting and linter compliance);
- [x] I have performed a self-review of my own code;
- [x] I have made corresponding changes to the documentation;
- [x] I have added tests that prove my fix is effective or that my feature works;
- [x] New and existing unit tests pass locally with my changes;
- [x] Add labels to distinguish the type of pull request. Available labels are `bug`, `enhancement`, `feature`, and `review`.

### CODE REVIEW

**Review by roelschr — COMMENTED** [review]
image

**Comment by AlvaroMarquesAndrade** [comment]
> image
> 
> 
> vai dar um trampinho aqui pra revisar auiehuiaeh mas boa parte disso já foi revisado em prs separados, certo? é mais procurar por possíveis breaking changes?

Opa, isso aí @roelschr. Vlw pela disponibilidade!

File: butterfree/hooks/schema_compatibility/cassandra_table_schema_compatibility_hook.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""Cassandra table schema compatibility Hook definition."""
```
Comment (AlvaroMarquesAndrade) [inline butterfree/hooks/schema_compatibility/cassandra_table_schema_compatibility_hook.py:1]: Considering that we're working on migrations' automation, I think we won't need these hooks anymore in the future. For now, we can keep them, but I think we should remove these schema compatibility hooks in the near future. What do you think?

Comment (moromimay) [reply butterfree/hooks/schema_compatibility/cassandra_table_schema_compatibility_hook.py:1]: I think so :thinking: :thought_balloon:

**Review by acsilveira — DISMISSED** [review]
LGTM.

**Review by acsilveira — APPROVED** [review]

**Pull request merged**